diff --git a/build/createIconSet.js b/build/createIconSet.js
--- a/build/createIconSet.js
+++ b/build/createIconSet.js
@@ -87,10 +87,14 @@
             if (!this.state.fontIsLoaded) {
                 return React.createElement(Text);
             }
-            return React.createElement(RNVIconComponent, {
-                ref: (view) => {
-                    this._icon = view;
-                },
-                ...this.props
-            });
+            var iconProps = {};
+            for (var k in this.props) {
+                if (this.props.hasOwnProperty(k)) iconProps[k] = this.props[k];
+            }
+            iconProps.ref = function(view) { this._icon = view; };
+            return React.createElement(RNVIconComponent, iconProps);
         }
     };
 }
